UndertaleModTool/Scripts/Community Scripts/TouchControlsEnabler.csx

```diff
@@ -100,11 +102,35 @@ ImportGMLFile(Path.Combine(dataPath, "gml_Object_obj_mobilecontrols_Step_0.gml")
 var mobileControls = Data.GameObjects.ByName("obj_mobilecontrols");
 mobileControls.Persistent = true;
 
-if (displayName == "deltarune chapter 1 & 2" || displayName == "deltarune chapter 1&2")
+var obj_time = Data.GameObjects.ByName("obj_time");
+if (obj_time is not null)
+{
+    Data.Code.ByName("gml_Object_obj_time_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
+    return;
+}
+var obj_gamecontroller = Data.GameObjects.ByName("obj_gamecontroller");
+if (obj_gamecontroller is not null)
 {
     Data.Code.ByName("gml_Object_obj_gamecontroller_Create_0").AppendGML("instance_create(0, 0, obj_mobilecontrols);", Data);
+    return;
 }
-else if (displayName == "undertale") 
+
+var firstRoom = Data.Rooms[0];
+var shouldAdd = true;
+
+foreach(var room_obj in firstRoom.GameObjects) {
+    if(room_obj.ObjectDefinition == mobileControls) {
+        shouldAdd = false;
+    }
+}
```

Comment:
You can use a Linq expression for this: `var shouldAdd = firstRoom.GameObjects.Any(o => o.ObjectDefinition == mobileControls);`